src/compiler/compile/utils/__test__.ts

```diff
@@ -9,8 +21,89 @@ describe('get_name_from_filename', () => {
 	it('uses the directory name, if basename is index', () => {
 		assert.equal(get_name_from_filename('path/to/Widget/index.svelte'), 'Widget');
 	});
+});
+
+describe('contenteditable', () => {
+	describe('isContenteditable', () => {
+		it('returns false if node is input', () => {
+			const node = { name: 'input' } as Element;
+			assert.equal(isContenteditable(node), false);
+		});
+		it('returns false if node is textarea', () => {
+			const node = { name: 'textarea' } as Element;
+			assert.equal(isContenteditable(node), false);
+		});
+		it('returns false if node is not input or textarea AND it is not contenteditable', () => {
+			const attr = { name: 'href' } as Attribute;
+			const node = { name: 'a', attributes: [attr] } as Element;
+			assert.equal(isContenteditable(node), false);
+		});
+		it('returns true if node is not input or textarea AND it is contenteditable', () => {
+			const attr = { name: CONTENTEDITABLE_ATTR } as Attribute;
+			const node = { name: 'a', attributes: [attr] } as Element;
+			assert.equal(isContenteditable(node), true);
+		});
+	});
+
+	describe('isInputOrTextarea', () => {
+		it('returns true if node is input', () => {
+			const node = { name: 'input' } as Element;
+			assert.equal(isInputOrTextarea(node), true);
+		});
+		it('returns true if node is textarea', () => {
+			const node = { name: 'textarea' } as Element;
+			assert.equal(isInputOrTextarea(node), true);
+		});
+		it('returns false if node is not input or textarea', () => {
+			const node = { name: 'div' } as Element;
+			assert.equal(isInputOrTextarea(node), false);
+		});
+	});
 
-	it('handles unusual filenames', () => {
```

Comment:
These deleted lines need to be restored as part of the `get_name_from_filename` test group.

Reply:
See https://github.com/sveltejs/svelte/pull/4262, where I address this issue directly (because this test fails and wasn't even testing what it was initially written for).

Reply:
Hmm, I don't think the change over on that one is quite right either (I'll add a comment over there).

The issue here is that you changed the same lines in 2 PRs which will create a conflict in whichever one is merged second.  Maybe that's unavoidable in this case.

Reply:
Yep there might be a conflict, which I don't mind resolving 🙂 I figured #4262 would probably be merged first anyway, since it's a lot simpler, plus I still need to add runtime test cases and address your other comments in this PR. 👍